source/corvusoft/restbed/service.cpp

```diff
@@ -176,29 +176,32 @@ namespace restbed
         
         if ( m_pimpl->m_ready_handler not_eq nullptr )
         {
-            m_pimpl->m_io_service->post( m_pimpl->m_ready_handler );
+            m_pimpl->get_io_service()->post( m_pimpl->m_ready_handler );
         }
         
         m_pimpl->m_uptime = steady_clock::now( );
         unsigned int limit = m_pimpl->m_settings->get_worker_limit( );
         
-        if ( limit > 0 )
+        if ( m_pimpl->m_external_io_service == nullptr )
```

Comment:
Currently I host WebServer in separate thread with separate io_service. But I believe this solution will scale for multiple threads by hosting each WebServer with io_service in a new thread using reuse_address(true).